app/scripts/lib/notification-manager.js

```diff
@@ -71,13 +74,16 @@ export default class NotificationManager extends EventEmitter {
       if (popupWindow.left !== left && popupWindow.state !== 'fullscreen') {
         await this.platform.updateWindowPosition(popupWindow.id, left, top);
       }
+      // pass new created popup window id to appController setter
+      // and store the id to private variable this._popupId for future access
+      this._setCurrentPopupId(popupWindow.id);
       this._popupId = popupWindow.id;
     }
   }
 
   _onWindowClosed(windowId) {
     if (windowId === this._popupId) {
-      this._popupId = undefined;
+      this._setCurrentPopupId(undefined);
```

Comment:
Can you help me understand what is different here from the method above that makes it so we only need to use `_setCurrentPopupId` and not assign it to the private variable `this._popupId = undefined`, where on `showPopup` we do the two simultaneously?

Reply:
yeah sure ! in this case i wanna trigger `setCurrentPopupId` in appState to clear the currentPopupId. And when we trigger the popup again, we will use the value from `getState` to re-examine `this._popupId`. But since you pointed out, i feel it is safer to clear local variable as well to keep consistent, and in case any other scenario i didn't catch up that would cause regression any more.